### ISSUE TITLE
MD051 anchor-style auto-default (flavor = "mkdocs" → python-markdown) does not apply under per-file-flavor

### ISSUE THREAD

**Issue by pwretmo** [rvben/rumdl#793]
### rumdl version

0.2.43

### Bug description

`docs/md051.md` documents that MD051's anchor-style auto-defaults to `python-markdown` when the effective flavor is `mkdocs`. That works when `mkdocs` is the *global* flavor or passed via `--flavor`, but **not** when it's set via `[per-file-flavor]` for a path pattern — MD051 keeps using `github`-style anchors (which double-hyphenate a dash-class character surrounded by spaces) for those files instead.

### Steps to reproduce

`docs/test.md`:

```markdown
### Getting Started — Advanced configuration (optional; only if defaults are insufficient)

[link A (single hyphen)](#getting-started-advanced-configuration-optional-only-if-defaults-are-insufficient)
[link B (double hyphen)](#getting-started--advanced-configuration-optional-only-if-defaults-are-insufficient)
```

1. No config (`rumdl check docs/test.md`): flags **link A** — `github`-style anchor computed, as expected for the default flavor.
2. `rumdl check --flavor mkdocs docs/test.md`: flags **link B** — anchor-style correctly auto-defaults to `python-markdown`, matching the documented behavior.
3. With this `.rumdl.toml`:

```toml
[global]
flavor = "gfm"

[per-file-flavor]
"docs/**/*.md" = "mkdocs"
```

   `rumdl check docs/test.md`: flags **link A** — identical to case 1, even though `docs/test.md` matches the `mkdocs` per-file-flavor pattern.

### Expected behavior

Case 3 should behave like case 2: MD051's anchor-style auto-default should read the *effective* (per-file-resolved) flavor for each file, matching how every other flavor-aware rule already behaves under `per-file-flavor` — not only the global/CLI flavor. Flagging link B (and treating link A as valid) is the correct result once `docs/test.md` is resolved to the `mkdocs` flavor.

### Actual behavior

Case 3 flags link A (the anchor that's actually correct under the `python-markdown`/mkdocs anchor-style) and silently accepts link B (which doesn't exist under that anchor-style) — identical output to having no `mkdocs` flavor active at all.

### Impact

Any project that (correctly, per the docs) uses `per-file-flavor` to scope `mkdocs` to its actual MkDocs-rendered docs tree while keeping other paths (README, GitHub-rendered files) on `gfm`/`github` gets silently wrong MD051 results for any heading containing a dash-class character surrounded by spaces (em dash, en dash, or a plain ` - `) in the `per-file-flavor`-scoped files — with no workaround, since `[MD051] anchor-style` is a single global setting and can't be scoped per-pattern the way `per-file-flavor` scopes everything else.

### Related issues

- #39 ("Two bugs in Links [MD051]") is the same defect *class* — a punctuation character surrounded by spaces mis-slugified — for the ASCII `-`/kramdown case; closed as fixed in 0.0.119, but verified against GitHub/Kramdown tooling only, not against `per-file-flavor` + `python-markdown` anchor-style.
- #481 ("Underscores stripped from heading slugs") is the same recurring pattern (slugifier disagrees with the target renderer for a specific character class); also closed as fixed.

### Configuration

```toml
[global]
flavor = "gfm"

[per-file-flavor]
"docs/**/*.md" = "mkdocs"
```

### Operating system

Windows (reproduced with pure string/config logic, so expected to be platform-independent)

**Comment by rvben** [comment]
Fixed in [v0.2.51](https://github.com/rvben/rumdl/releases/tag/v0.2.51).

You had the cause right in the report. An unpinned `anchor-style` is derived from the flavor, because the anchor a heading gets is whatever its renderer emits - but that derivation ran once at rule construction, where only the global flavor is known. A file `per-file-flavor` handed to another flavor was still checked against the global flavor's anchors. The style is now settled when a file is in hand, from the flavor that file is parsed with. A style you configure explicitly is still yours and still applies to every file, so `anchor-style` keeps overriding the flavor exactly as before.

MD080 had the same defect from the same line and is fixed with it: heading-anchor collisions that only exist under a file's own anchor style were being missed.

Your three cases on v0.2.51, run against your fixture exactly as written (`--enable MD051` only, to keep the unrelated MD041 the fixture also trips out of the way):

```console
$ rumdl check --no-cache --no-config --enable MD051 docs/test.md
docs/test.md:3:1: [MD051] Link anchor '#getting-started-advanced-configuration-optional-only-if-defaults-are-insufficient' does not exist in document headings

$ rumdl check --no-cache --no-config --flavor mkdocs --enable MD051 docs/test.md
docs/test.md:4:1: [MD051] Link anchor '#getting-started--advanced-configuration-optional-only-if-defaults-are-insufficient' does not exist in document headings

$ rumdl check --no-cache --enable MD051 docs/test.md   # [global] flavor = "gfm" + [per-file-flavor] "docs/**/*.md" = "mkdocs"
docs/test.md:4:1: [MD051] Link anchor '#getting-started--advanced-configuration-optional-only-if-defaults-are-insufficient' does not exist in document headings
```

Case 3 now matches case 2, and case 1 still flags link A, which is the control proving the flavor is what changes the answer rather than the check having been loosened. Adding `[MD051] anchor-style = "github"` to that same config puts case 3 back on line 3, which is the other control: an explicit style still wins over the flavor.

MD080 behaves the same way now. Two headings whose anchors collide only under the python-markdown style, in one file under `docs/` and an identical copy outside it:

```console
$ rumdl check --no-cache --enable MD080 docs/collide.md outside.md
docs/collide.md:3:3: [MD080] Heading anchor 'foo-bar' collides with the heading at line 1; fragment links and any derived page identifier resolve only to the first occurrence
```

Only the file the pattern claims is read as mkdocs.

Thank you for the report - the impact section was accurate and it is the part that made this urgent: silently wrong results with no workaround, since `[MD051] anchor-style` is global and cannot be scoped the way `per-file-flavor` scopes everything else.
